src/tools/annotation/CircleRoiTool.js

```diff
@@ -497,16 +509,44 @@ function _calculateStats(image, element, handles, modality, pixelSpacing) {
 
   if (modality === 'PT') {
     meanStdDevSUV = {
-      mean: calculateSUV(image, ellipseMeanStdDev.mean, true) || 0,
-      stdDev: calculateSUV(image, ellipseMeanStdDev.stdDev, true) || 0,
+      mean:
+        measurementUncertainty.getGenericRounding(
+          calculateSUV(image, ellipseMeanStdDev.mean, true)
+        ) || 0,
+      stdDev:
+        measurementUncertainty.getGenericRounding(
+          calculateSUV(image, ellipseMeanStdDev.stdDev, true)
+        ) || 0,
     };
   }
 
   const radius =
     (circleCoordinates.width *
       ((pixelSpacing && pixelSpacing.colPixelSpacing) || 1)) /
     2;
+
   const perimeter = 2 * Math.PI * radius;
+
+  const pixelDiagonal = measurementUncertainty.getPixelDiagonal(
```

Comment:
From this point on, I would suggest considering extracting logic into functions. Is there any logic that is duplicated in other files like RectangleRoiTool i.e.?

I would try to fit into small functions these transformations and see which of them could be centralized in measurementUncertaintyTool.js.

It seems that some operations like instantiating a Decimal object and || 0 are applied twice to the same value here.